### ISSUE TITLE
Whatsnew files wihtout a country are not detected

### ISSUE THREAD

**Issue by gabriprat** [bitrise-steplib/steps-google-play-deploy#48]
My app has three locales at Google Play: `en-US`, `es-ES` and `ca`.

I have 3 whatsnew files called:
* whatsnew-ca
* whatsnew-en-US
* whatsnew-es-ES

The first one is not detected.

**Comment by fehersanyi-bitrise** [comment]
Hello there, could you please send me the latest build's url with this issue, so I can check the logs for some more details, and try to come up with a possible solution? 🙃

**Comment by pamartineza** [comment]
@gabriprat same issue here
@fehersanyi-bitrise I believe the problem is the regex pattern **whatsnew-(?P<local>.\*-.\*)** used to get the locale, it just accepts language files that include regions, but Catalan as many other languages in Google Play Listing has not ISO region, it is just "ca"



> 

```golang
func readLocalisedRecentChanges(recentChangesDir string) (map[string]string, error) {
	recentChangesMap := map[string]string{}
	pattern := filepath.Join(recentChangesDir, "whatsnew-*-*")
	recentChangesPaths, err := filepath.Glob(pattern)
	if err != nil {
		return map[string]string{}, err
	}

	pattern = `whatsnew-(?P<local>.*-.*)`
	re := regexp.MustCompile(pattern)

	for _, recentChangesPath := range recentChangesPaths {
		matches := re.FindStringSubmatch(recentChangesPath)
		if len(matches) == 2 {
			lanugage := matches[1]
			content, err := fileutil.ReadStringFromFile(recentChangesPath)
			if err != nil {
				return map[string]string{}, err
			}

			recentChangesMap[lanugage] = content
		}
	}

	return recentChangesMap, nil
}
```

**Comment by pamartineza** [comment]
Hi, @fehersanyi-bitrise any expected ETA here? I was told by Bence from support chat that a task to solve it has been already created, but almost a month later there is still no fix, no news

**Comment by gabriprat** [comment]
I agree on the problem description by @pamartineza . If you need a build URL check this https://app.bitrise.io/build/e5a90f8aa09a027d
Can we do something more to help you solve the issue?

**Comment by ahvth-bitrise** [comment]
Hi @gabriprat @pamartineza . Our tooling team is currently aware of the issue and a fix should hopefully be available soon. Thanks for your feedback and your patience! 😄

**Comment by gabriprat** [comment]
Hi, any news of the fix? It seems as easy as changing the regex pattern as suggested in https://github.com/bitrise-steplib/steps-google-play-deploy/issues/48#issuecomment-505440638

**Comment by lpusok** [comment]
Hi @gabriprat, @pamartineza
Thanks for your patience, there is a new version of the step available (3.0.1 - https://github.com/bitrise-steplib/steps-google-play-deploy/releases/tag/3.0.1) which fixes this issue. Please let us know if you encounter any issues with it.

**Comment by lpusok** [comment]
Closing as fix is released.
